tests/test_marshal.py

```diff
@@ -0,0 +1,40 @@
+import marshal
+import unittest
+
+from hypothesis import example, given, strategies as st
+
+simple_immutables = (
+    st.integers()
+    | st.booleans()
+    | st.floats(allow_nan=False)
+    | st.complex_numbers(allow_nan=False)
```

Comment:
I'd comment for later readers that `nan` marshals just fine, but we exclude it to make asserting equality easier.

Reply:
good point, have pushed a comment